cardano-cli/src/Cardano/CLI/Json/Friendly.hs

```diff
@@ -293,6 +297,16 @@ friendlyTxBodyImpl
     friendlyLedgerProposals cOnwards proposalProcedures =
       Array $ fromList $ map (friendlyLedgerProposal cOnwards) proposalProcedures
 
+-- | API doesn't yet show that supplemental datums are alonzo onwards. So we do it in this prototype,
+-- even if we don't use the witness.
+friendlySupplementalDatums
```

Comment:
This is an important function to review.

Reply:
We should render the datum hashes as well. See my other comment.

Reply:
Changed from:

```hs
instance ToJSON HashableScriptData where
  toJSON = scriptDataToJsonDetailedSchema
```

to:

```hs
instance ToJSON HashableScriptData where
  toJSON hsd =
    object
      [ "bytes" .= (String $ Text.decodeUtf8 $ getOriginalScriptDataBytes hsd)
      , "json" .= scriptDataToJsonDetailedSchema hsd
      ]
```

Reply:
No this is not that useful. We want the hash otherwise users will need to calculate the hash themselves.

Reply:
Ah sorry I didn't understood that the first time. Did this change:

```diff
@@ -29,6 +28,6 @@ instance (L.EraTxOut ledgerera, L.EraGov ledgerera) => ToJSON (L.NewEpochState l
 instance ToJSON HashableScriptData where
   toJSON hsd =
     object
-      [ "bytes" .= String (Text.decodeUtf8 $ getOriginalScriptDataBytes hsd)
+      [ "hash" .= hashScriptDataBytes hsd
       , "json" .= scriptDataToJsonDetailedSchema hsd
       ]
```